server/config.js

```diff
@@ -48,6 +56,6 @@ export function loadConfig (phase: string, dir: string, customConfig?: ?Object)
   return withDefaults(userConfig)
 }
 
-function withDefaults (config: Object) {
+function withDefaults (config: {} | NextConfig) {
```

Comment:
why `{} | NextConfig`. Everything in `NextConfig` is a Maybe type already, meaning an empty object can already be a `NextConfig` type

Reply:
Nope, that's a lie. But in any case, `{} | NextConfig` is the same thing as `{}` which can accept ANY OBJECT. That's pretty broad. Can we fix this?

Reply:
I feel like (https://flow.org/en/docs/types/utilities/#toc-subtype) will be useful, but it's WIP. maybe one of the other utilities?